test/jdk/sun/security/pkcs11/PKCS11Test.java

```diff
@@ -536,36 +539,50 @@ public static void useNSS() {
         nss_library = "nss3";
     }
 
+    // Run NSS testing on the static PKCS11 test Provider
     public static void testNSS(PKCS11Test test) throws Exception {
+        testNSS(test, null);
+    }
+
+    // Run NSS testing on a Provider p configured with test nss config
+    public static void testNSS(PKCS11Test test, Provider p) throws Exception {
+        String nssConfig = getNssConfig();
+        if (nssConfig == null) {
+            // issue loading libraries
+            return;
+        }
+        p = getSunPKCS11(nssConfig);
+        test.premain(p);
```

Comment:
It seems a bit strange that the 2nd argument p isn't really used, but rather overridden with getSunPKCS11(String config) call. Do you mean to call getSunPKCS11(nssConfig, p) here if p is non-null?